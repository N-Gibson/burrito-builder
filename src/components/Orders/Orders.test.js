import React from 'react';
import { Orders } from './Orders';
import { shallow } from 'enzyme';
import { mapStateToProps, mapDispatchToProps } from './Orders';
import { setOrders } from '../../actions/index';

describe('Orders', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Orders orders={[{ name: 'Bob', ingredients: ['eggs', 'potato'], id: 1}, { name: 'Sue', ingredients: ['pancake', 'sausage'], id: 2}]}/>)
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Orders mapStateToProps', () => {
  const mockState = {
    orders: [{ name: 'Bob', ingredients: ['eggs', 'potato']}, { name: 'Sue', ingredients: ['pancake', 'sausage']}]
  }

  const expectedState = {
    orders: [{ name: 'Bob', ingredients: ['eggs', 'potato']}, { name: 'Sue', ingredients: ['pancake', 'sausage']}]
  }

  const mappedProps = mapStateToProps(mockState);
  expect(mappedProps).toEqual(expectedState);
});

describe('Orders mapDispatchToProps', () => {
  let mockDispatch, mappedDispatch;

  beforeEach(() => {
    mockDispatch = jest.fn();
    mappedDispatch = mapDispatchToProps(mockDispatch);
  });

  it('setOrders', () => {
    const actionToDispatch = setOrders([{ name: 'Bob', ingredients: ['eggs', 'potato']}, { name: 'Sue', ingredients: ['pancake', 'sausage']}]);
    mappedDispatch.setOrders([{ name: 'Bob', ingredients: ['eggs', 'potato']}, { name: 'Sue', ingredients: ['pancake', 'sausage']}])

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });
});