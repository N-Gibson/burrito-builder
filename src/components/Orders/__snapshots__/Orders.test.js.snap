// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Orders should match snapshot 1`] = `
<section>
  <div
    className="order"
    key="1"
  >
    <h3>
      Bob
    </h3>
    <ul
      className="ingredient-list"
    >
      <li
        key="eggs"
      >
        eggs
      </li>
      <li
        key="potato"
      >
        potato
      </li>
    </ul>
  </div>
  <div
    className="order"
    key="2"
  >
    <h3>
      Sue
    </h3>
    <ul
      className="ingredient-list"
    >
      <li
        key="pancake"
      >
        pancake
      </li>
      <li
        key="sausage"
      >
        sausage
      </li>
    </ul>
  </div>
</section>
`;
